### PR TITLE
fixes / first tests (waiting for P3)

### PR DESCRIPTION
**Pull request by Wwarrior1**

### CODE REVIEW

File: hypergraphs/tests/test_p5_p6.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        P1(self.graph, x_max_idx=width - 1, y_max_idx=height - 1, image=self.image)
+
+        hyperedges_to_remove = [x for x, y in self.graph.nodes(data=True)
+                                if 'label' in y.keys() and y['label'] == HyperEdge.B.name]
+        for id in hyperedges_to_remove:
+            self.graph.remove_node(id)
+
+        hyperedge = [(x, y) for x, y in self.graph.nodes(data=True)
+                     if 'label' in y.keys() and y['label'] == HyperEdge.I.name][0]
+        hyperedge[1]['should_break'] = 0
```
Comment (Rockfan121) [inline hypergraphs/tests/test_p5_p6.py]: In P1 'should_break'  seems to be set to 0 by default (

Comment (Rockfan121) [inline hypergraphs/tests/test_p5_p6.py]: In P1 'should_break' seems to be set to 0 by default

File: hypergraphs/utils.py [hunk 2]
```diff
@@ -12,25 +20,30 @@ def get_node_id(node_position: Tuple[int, int]) -> int:
...
 def get_i_nodes(graph, common_node_id):
-    if graph.node[hyperedge_id]['is_hyperedge']:
+    return __get_x_nodes(graph, common_node_id, HyperEdge.I)
+
+
+def __get_x_nodes(graph, common_node_id, label):
+    if graph.node[common_node_id]['is_hyperedge']:
         raise ValueError('Given node_id is not id of common node')
-    return [i_node for i_node in graph[common_node_id] if graph.nodes[f1_node]['label'] == 'I']
+    return [x_node for x_node in graph[common_node_id] if graph.nodes[x_node]['label'] == label]
```
Comment (Rockfan121) [inline hypergraphs/utils.py]: return [x_node for x_node in graph[common_node_id] if graph.nodes[x_node]['label'] == label.name]

**Review by Rockfan121 — APPROVED** [review]
Two lines I've commented probably need fixing. Everything else looks great :-)

**Comment by Callisto210** [comment]
Has someone tested if algorithm in P6 works? Actually P3 is implemented on separate branch, I think we can rebase on top of it, and use it. Definitions are unlikely to be changed, even if P3 is not working, we will have code that uses it.

Good job! Code looks awesome!

Thanks for fixing mistakes in comments ;D

**Pull request merged**